api/environments/component_handler.go

```diff
@@ -126,6 +131,64 @@ func (eh EnvironmentHandler) RestartComponent(appName, envName, componentName st
 	return nil
 }
 
+// RestartComponentAuxiliaryResource Restarts a component's auxiliary resource
+func (eh EnvironmentHandler) RestartComponentAuxiliaryResource(appName, envName, componentName, auxType string) error {
+	log.Infof("Restarting auxiliary resource %s for component %s, %s", auxType, componentName, appName)
+
+	deploySummary, err := eh.deployHandler.GetLatestDeploymentForApplicationEnvironment(appName, envName)
+	if err != nil {
+		return err
+	}
+
+	deploymentDto, err := eh.deployHandler.GetDeploymentWithName(appName, deploySummary.Name)
+	if err != nil {
+		return err
+	}
+
+	componentDto := deploymentDto.GetComponentByName(componentName)
+	if componentDto == nil {
+		return environmentModels.NonExistingComponent(appName, componentName)
+	}
+
+	auxResourceDto := componentDto.GetAuxiliaryResourceByType(auxType)
+	if auxResourceDto == nil {
+		return environmentModels.NonExistingComponentAuxiliaryType(appName, componentName, auxType)
+	}
+
+	if auxResourceDto.Deployment == nil || auxResourceDto.Deployment.Status != deploymentModels.ConsistentComponent.String() {
+		currentStatus := deploymentModels.ComponentReconciling.String()
+		if auxResourceDto.Deployment != nil {
+			currentStatus = auxResourceDto.Deployment.Status
+		}
+		return environmentModels.CannotRestartComponent(appName, componentName, currentStatus)
+	}
+
+	selector := labelselector.ForAuxiliaryResource(appName, componentName, auxType).String()
```

Comment:
Can the logic just start from here? With error `environmentModels.NonExistingComponent(appName, componentName)` if the deployment does not exist?
I tried to watch deployment status `k get deploy  -n app-namespace www -o json|jq -r .status`
while running commands
`k delete pods --all -n app-namespace`
`k rollout restart deployment www  -n app-namespace`
Status was changed , but a bit unclear way. So I'm not sure if it is good idea.